x-pack/plugins/lens/public/datatable_visualization/expression.tsx

```diff
@@ -214,6 +264,120 @@ export function DatatableComponent(props: DatatableRenderProps) {
     return <EmptyPlaceholder icon={LensIconChartDatatable} />;
   }
 
+  const tableColumns: Array<EuiBasicTableColumn<unknown>> = props.args.columns.columnIds
```

Comment:
This should also be moved over to the improved type

Reply:
Done.